src/main/java/duckling/Server.java

```diff
@@ -1,11 +1,101 @@
 package duckling;
 
+import duckling.requests.Request;
+import duckling.requests.RequestBuilder;
+import duckling.responders.Responders;
+
+import java.io.*;
+import java.net.*;
+import java.util.concurrent.*;
+
 public class Server {
+    public static final String CLRF = "\r\n";
+
     public int port;
+    public String root;
+
+    private boolean shuttingDown = false;
```

Comment:
That's really on the nose, I had some issues with this while putting it together for (what I'm guessing are) exactly these reasons.